homeassistant/components/harmony/remote.py

```diff
@@ -110,44 +114,14 @@ async def async_setup_entry(
     _LOGGER.debug("Harmony Remote: %s", device)
 
     async_add_entities([device])
-    register_services(hass)
-
-
-def register_services(hass):
-    """Register all services for harmony devices."""
-
-    async def _apply_service(service, service_func, *service_func_args):
-        """Handle services to apply."""
-        entity_ids = service.data.get("entity_id")
-
-        want_devices = [
-            hass.data[DOMAIN][config_entry_id] for config_entry_id in hass.data[DOMAIN]
-        ]
 
-        if entity_ids:
-            want_devices = [
-                device for device in want_devices if device.entity_id in entity_ids
-            ]
+    platform = entity_platform.current_platform.get()
 
-        for device in want_devices:
-            await service_func(device, *service_func_args)
-
-    async def _sync_service(service):
-        await _apply_service(service, HarmonyRemote.sync)
-
-    async def _change_channel_service(service):
-        channel = service.data.get(ATTR_CHANNEL)
-        await _apply_service(service, HarmonyRemote.change_channel, channel)
-
-    hass.services.async_register(
-        DOMAIN, SERVICE_SYNC, _sync_service, schema=HARMONY_SYNC_SCHEMA
+    platform.async_register_entity_service(
+        SERVICE_SYNC, HARMONY_SYNC_SCHEMA, "sync",
```

Comment:
If we pass a dict for the schema instead of a voluptuous schema, the helper will create an entity service schema for us.
https://github.com/home-assistant/core/blob/f492a7678d139df564bad4453646b23262dfab88/homeassistant/helpers/config_validation.py#L790-L806

Reply:
Very nice,  I'll reduce this on the next harmony update.